packages/react-input/src/components/Input/Input.types.ts

```diff
@@ -20,12 +20,21 @@ export type InputSlots = {
   insideEnd?: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
 };
 
+export interface InputCommons {
+  /** @default 'medium' */
+  // TODO this overlaps with a native input prop
+  size?: 'small' | 'medium' | 'large';
+  inline?: boolean;
+  /** @default 'outline' */
+  appearance?: 'filledDarker' | 'filledLighter' | 'underline' | 'outline';
+}
+
 /**
  * Input Props
  */
-export interface InputProps extends ComponentProps<Partial<InputSlots>> {}
```

Comment:
Question - when moving to React hooks, many teams choose to move from interface to type declarations for their props and use type intersections over extends.  Are interfaces giving us something that types would not?  When I see interfaces I always think a class will implement it.

Reply:
IIRC we discussed this but didn't come to a solid conclusion about which convention we wanted (and the discussion apparently happened before we started documenting things in RFCs). There *may* be some specific reason that `type` is commonly used for slots, but I can't remember offhand. @layershifter do you remember?

Reply:
> There _may_ be some specific reason that `type` is commonly used for slots, but I can't remember offhand. @layershifter do you remember?

When TS emits type declarations _sometimes_ it inlines `type` declarations (https://github.com/microsoft/TypeScript/issues/37151), so in general `interfaces` are preferred for in our case.

> We have seen extreme cases where this duplication has inflated declaration files from **7KB to 700KB**. That’s quite a lot of redundant code to download and parse.
>
> Through experimentation, we discovered potential techniques to prevent inlining of type declarations, such as:
>
>- prefer `interface` instead of `type` (interfaces are not inlined)
>- If an interface needed by a declaration is not exported, tsc will refuse to inline the type and will generate a clear error (e.g., `TS4023: Exported variable has or is using name from external module but cannot be named.`)

https://www.techatbloomberg.com/blog/10-insights-adopting-typescript-at-scale/

Reply:
Interesting, thanks! Incidentally this diff (the old version) shows an example of expanded types. 😬 https://github.com/microsoft/fluentui/pull/19376/files#diff-805da48a70295edd847bed2a0dda92910c21f9f0185fa238bc342d167ae4407b